docs/src/FAQ.md

```diff
@@ -76,3 +76,52 @@ ChainRulesTestUtils.jl has some dependencies, so it is a separate package from C
 This means your package can depend on the light-weight ChainRulesCore.jl, and make ChainRulesTestUtils.jl a test-only dependency.
 
 Remember to read the section on [On writing good `rrule` / `frule` methods](@ref).
+
+## How do chain rules work for complex functions?
+
+`ChainRules.jl` follows the convention that `frule` applied to a function ``f(x + i y) = u(x,y) + i \, v(x,y)`` with perturbation ``\dot x + i \dot y`` returns the value and 
+```math
+\tfrac{\partial u}{\partial x} \, \dot x + \tfrac{\partial u}{\partial y} \, \dot y + i \, \Bigl( \tfrac{\partial v}{\partial x} \, \dot x + \tfrac{\partial v}{\partial y} \, \dot y \Bigr)
+,
+```
+and similarly `rrule` applied to the same function evaluates the function and returns a pullback which, when applied to adjoint ``\bar u + i \bar v``, returns
+```math
+\bar u \, \tfrac{\partial u}{\partial x} + \bar v \, \tfrac{\partial v}{\partial x} + i \, \Bigl(\bar u \, \tfrac{\partial u }{\partial y} + \bar v \, \tfrac{\partial v}{\partial y} \Bigr)
+.
+```
+Note that if we interpret complex numbers as vectors in ``\mathbb{R}^2``, then these rules correspond to left- and right-multiplication with the Jacobian of ``f(z)``, i.e. `frule` corresponds to 
+```math
+\begin{pmatrix}
+\tfrac{\partial u}{\partial x} \, \dot x + \tfrac{\partial u}{\partial y} \, \dot y 
+\\ 
+\tfrac{\partial v}{\partial x} \, \dot x + \tfrac{\partial v}{\partial y} \, \dot y 
+\end{pmatrix}
+=
+\begin{pmatrix}
+\tfrac{\partial u}{\partial x} & \tfrac{\partial u}{\partial y} \\
+\tfrac{\partial v}{\partial x} & \tfrac{\partial v}{\partial y} \\
+\end{pmatrix}
+\begin{pmatrix}
+\dot x \\ \dot y
+\end{pmatrix}
+
+```
+and `rrule` corresponds to 
+```math
+\begin{pmatrix}
+\bar u \, \tfrac{\partial u}{\partial x} + \bar v \, \tfrac{\partial v}{\partial x} 
+& 
+\bar u \, \tfrac{\partial u}{\partial y} + \bar v \, \tfrac{\partial v}{\partial y} 
+\end{pmatrix}
+=
+\begin{pmatrix}
+\bar u & \bar v
+\end{pmatrix}
+\begin{pmatrix}
+\tfrac{\partial u}{\partial x} & \tfrac{\partial u}{\partial y} \\
+\tfrac{\partial v}{\partial x} & \tfrac{\partial v}{\partial y} \\
+\end{pmatrix}
+.
+```
+
+If ``f(z)`` is holomorphic, then `frule` can be implemented as ``f'(z) \, \dot z`` and `rrule` can be implemented as ``\bar f \,  \overline{f'(z))}``. 
```

Comment:
Since every non-trivial holomorphic function is `C -> C`, I think it's important to emphasize here that this is valid for intermediate steps, but if your input/output (depending on whether it's f-/rrule) is not real, this doesn't really give you a meaningful answer on its own.

Reply:
Correct. As pointed out above, we should provide another layer on top of `ChainRules` to provide a more reasonable API for evaluating Wirtinger derivatives, Jacobians, gradients of functions `Complex -> Real`, etc.